app/core/voyagerUtils.js

```diff
@@ -0,0 +1,116 @@
+        levelDataMap[levelOriginal] = level.attributes || level
+      }
+    }
+    return levelDataMap
+  },
+
+  /**
+   * Gets the next level original ids for a given level
+   * @param {Object} level - The level object.
+   * @param {Object[]} level.nextLevels - The array of nextLevels for the given level.
+   * @param {string} level.nextLevels[].levelOriginal - Original id of the next level.
+   * @returns {string[]} - Array of next level 'Original' ids.
+   */
+  getNextLevelOriginalForLevel: function (level) {
+    const nextLevels = level.nextLevels
+    const nextLevelOriginals = []
+    if ((nextLevels || []).length > 0) {
+      nextLevelOriginals.push(nextLevels[0].levelOriginal) // assuming that there will be just one next level for voyager v1 as of now.
+      // TODO: handle logic for 1FH capstone level
+    }
+    return nextLevelOriginals
+  },
+
+  /**
+   * Calculates all the next levels for a list of levels in a classroom/campaign based on the level sessions.
+   * @param {Object[]} sessions - The list of level session objects.
+   * @param {Object[]|Object} levels - The list of level objects, or an object with keys as level original id and value as level data.
+   * @param {Object[]} levels.nextLevels - The array of nextLevels for a level.
+   * @param {string} levels.nextLevels[].levelOriginal - Original id of the next level.
+   * @param {Object} levels.position - The object containing position of a level.
+   * @param {boolean} levels.first - Value to determine if a level is the first level of classroom/campaign.
+   * @param {Object} levelStatusMap - Optional. Object with key as the level original id, and value as complete/started.
+   * @returns {string[]} - Array of next level original ids.
+
+   * There would only be 1 next level for voyager v1 as of now, hence use the first element of the array returned from here.
+   * TODO: Edge case - If the 'sessions' contain a session for a level which has been played in another classroom, or another campaign;
+   * then it will be part of the nextLevels even if not unlocked for the current classroom/campaign.
+   */
+  findNextLevelsBySession: function (sessions, levels, levelStatusMap) {
+    if (!levelStatusMap) {
+      levelStatusMap = this.getLevelStatusMap(sessions)
+    }
+    const nextLevelOriginals = new Set() // next level = started levels + incomplete unlocked levels + not-started first levels
+
+    let levelDataMap = {}
+    if (_.isArray(levels)) {
+      levelDataMap = this.getLevelDataMap(levels)
+    } else {
+      levelDataMap = levels
+    }
+    for (let [levelOriginal, level] of Object.entries(levelDataMap)) {
+      if (levelStatusMap[levelOriginal] === 'started') {
+        nextLevelOriginals.add(levelOriginal)
+      } else if (levelStatusMap[levelOriginal] === 'complete') {
+        const unlockedLevelOriginals = this.getNextLevelOriginalForLevel(level)
+        if (unlockedLevelOriginals.length > 0) { // incomplete unlocks
```

Comment:
This check may not be required. As long as `this.getNextLevelOriginalForLevel(level)` always returns an array. Or by adding an `|| []` after it. The `filter` and `forEach` will do nothing with an empty array.